src/query/execute_schema.c

```diff
@@ -7016,21 +7069,23 @@ do_promote_partition (SM_CLASS * class_)
       smattr->class_mop = subclass_mop;
     }
 
-  /* Make sure we do not copy anything that actually belongs to the root class (the class to which this partition
-   * belongs to). This includes: auto_increment flags, unique indexes, primary keys, and foreign keys */
+  /* Ensure that attributes belonging to a partition are not copied.  
+   * However, according to EPIC CBRD-26096, primary key (PK) constraints and 
+   * NOT NULL UNIQUE properties must be preserved and reflected properly. */
   for (smattr = ctemplate->attributes; smattr != NULL; smattr = (SM_ATTRIBUTE *) smattr->header.next)
     {
       /* reset flags that belong to the root partitioned table */
       smattr->auto_increment = NULL;
       smattr->flags &= ~(SM_ATTFLAG_AUTO_INCREMENT);
-      if ((smattr->flags & SM_ATTFLAG_PRIMARY_KEY) != 0)
+      if (need_reset_notnull_unique_flags (smattr))
```

Comment:
NIT:

검사 후 리셋하려는 대상이 uk이기 때문에 다음 네이밍이 더 적합해 보입니다. 또는 has_notnull_unique_constraints() 등을 추천합니다.
```suggestion
      if (need_reset_unique_flags (smattr))
```

Reply:
반영했습니다. has_notnull_unique_constraints() 로 변경되었습니다.